src/stores/GroupStoreCache.js

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2017 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import GroupStore from './GroupStore';
+
+class GroupStoreCache {
+    constructor() {
+        this.groupStores = {};
+    }
+
+    getGroupStore(matrixClient, groupId) {
+        if (!this.groupStores[groupId]) {
+            this.groupStores[groupId] = new GroupStore(matrixClient, groupId);
+        }
+        return this.groupStores[groupId];
```

Comment:
ah, so this is storing all the group stores that get made rather than just the latest? There's the concern about keeping them in memory, but mostly the thing here will be that re-opening the GroupView on a group you've previously viewed will get a cached GroupStore and therefore not cause a re-fetch of the group data, so you'll get a stale view of the group.